IBM/operand-deployment-lifecycle-manager
# Add support for sharing Route data via ConfigMap with OperandBindInfo

See commit history for details

How to test:
1. `export DEV_REGISTRY=quay.io/username`
1. `export QUAY_REGISTRY="$DEV_REGISTRY"`
1. `make build-push-dev-image`
1. edit manager.yaml image with the image you just built (ideally also change ImagePullPolicy to Always)
1. `make bundle-manifests`
1. `make build-push-bundle-image`
1. `make run-bundle`
1. Create template below for the various resources for testing
1. Ensure you have an OperandRegistry for im-operator
1. Verify that a configmap is created with route data

templates
```
kind: Route
apiVersion: route.openshift.io/v1
metadata:
  name: my-route
spec:
  host: my-route-henry-test.apps.henryhlibackup.cp.fyre.ibm.com
  to:
    kind: Service
    name: my-service
    weight: 100
  port:
    targetPort: 80
  tls:
    termination: passthrough
  wildcardPolicy: None
---
apiVersion: v1
kind: Service
metadata:
  name: my-service
  namespace: henry-test
spec:
  selector:
    app: MyApp
  ports:
    - protocol: TCP
      port: 80
      targetPort: 9376
---
apiVersion: operator.ibm.com/v1alpha1
kind: OperandBindInfo
metadata:
  labels:
    app.kubernetes.io/instance: operand-deployment-lifecycle-manager
    app.kubernetes.io/managed-by: operand-deployment-lifecycle-manager
    app.kubernetes.io/name: operand-deployment-lifecycle-manager
  name: example-service
  namespace: henry-test
spec:
  bindings:
    public-my-route:
      route:
        name: my-route
        data:
          HELLO: .spec.host
  description: Binding information that should be accessible to MongoDB adopters
  operand: ibm-im-operator
  registry: common-service
---
apiVersion: operator.ibm.com/v1alpha1
kind: OperandRequest
metadata:
  name: common-service
  namespace: henry-test
spec:
  requests:
  - operands:
    - name: ibm-im-operator
    registry: common-service
    registryNamespace: ibm-common-services
```
